hcloud/schema_gen.go

```diff
@@ -0,0 +1,875 @@
+
+	NetworkRouteFromSchema(schema.NetworkRoute) NetworkRoute
+
+	SchemaFromNetworkRoute(NetworkRoute) schema.NetworkRoute
+
+	LoadBalancerFromSchema(schema.LoadBalancer) *LoadBalancer
+
+	SchemaFromLoadBalancer(*LoadBalancer) schema.LoadBalancer
+
+	// goverter:map Prices Pricings
+	LoadBalancerTypeFromSchema(schema.LoadBalancerType) *LoadBalancerType
+
+	// goverter:map Pricings Prices
+	SchemaFromLoadBalancerType(*LoadBalancerType) schema.LoadBalancerType
+
+	// goverter:map PriceHourly Hourly
+	// goverter:map PriceMonthly Monthly
+	LoadBalancerTypeLocationPricingFromSchema(schema.PricingLoadBalancerTypePrice) LoadBalancerTypeLocationPricing
+
+	// goverter:map Hourly PriceHourly
+	// goverter:map Monthly PriceMonthly
+	SchemaFromLoadBalancerTypeLocationPricing(LoadBalancerTypeLocationPricing) schema.PricingLoadBalancerTypePrice
+
+	LoadBalancerServiceFromSchema(schema.LoadBalancerService) LoadBalancerService
+
+	SchemaFromLoadBalancerService(LoadBalancerService) schema.LoadBalancerService
+
+	LoadBalancerServiceHealthCheckFromSchema(*schema.LoadBalancerServiceHealthCheck) LoadBalancerServiceHealthCheck
+
+	SchemaFromLoadBalancerServiceHealthCheck(LoadBalancerServiceHealthCheck) *schema.LoadBalancerServiceHealthCheck
+
+	LoadBalancerTargetFromSchema(schema.LoadBalancerTarget) LoadBalancerTarget
+
+	SchemaFromLoadBalancerTarget(LoadBalancerTarget) schema.LoadBalancerTarget
+
+	// goverter:map ID Server
+	LoadBalancerTargetServerFromSchema(schema.LoadBalancerTargetServer) LoadBalancerTargetServer
+
+	// goverter:map Server ID
+	SchemaFromLoadBalancerServerTarget(LoadBalancerTargetServer) schema.LoadBalancerTargetServer
+
+	LoadBalancerTargetHealthStatusFromSchema(schema.LoadBalancerTargetHealthStatus) LoadBalancerTargetHealthStatus
+
+	SchemaFromLoadBalancerTargetHealthStatus(LoadBalancerTargetHealthStatus) schema.LoadBalancerTargetHealthStatus
+
+	CertificateFromSchema(schema.Certificate) *Certificate
+
+	SchemaFromCertificate(*Certificate) schema.Certificate
+
+	PaginationFromSchema(schema.MetaPagination) Pagination
+
+	SchemaFromPagination(Pagination) schema.MetaPagination
+
+	// goverter:ignore response
+	// goverter:ignore DetailsRaw
+	// goverter:map Details Details | errorDetailsFromSchema
+	ErrorFromSchema(schema.Error) Error
+
+	// goverter:ignore response
+	// goverter:map Details Details | schemaFromErrorDetails
+	// goverter:map Details DetailsRaw | rawSchemaFromErrorDetails
+	SchemaFromError(Error) schema.Error
+
+	// goverter:map . Image | imagePricingFromSchema
+	// goverter:map . FloatingIP | floatingIPPricingFromSchema
+	// goverter:map . FloatingIPs | floatingIPTypePricingFromSchema
+	// goverter:map . PrimaryIPs | primaryIPPricingFromSchema
+	// goverter:map . Traffic | trafficPricingFromSchema
+	// goverter:map . ServerTypes | serverTypePricingFromSchema
+	// goverter:map . LoadBalancerTypes | loadBalancerTypePricingFromSchema
+	// goverter:map . Volume | volumePricingFromSchema
```

Comment:
You could remove the need for this custom mapping method by defining two additional goverter methods:
```go
    // ... all other :map
	// goverter:map . Volume
    PricingFromSchema(schema.Pricing) Pricing
	// goverter:map . PerGBMonthly
	pricingToVolume(schema.Pricing) VolumePricing
	// goverter:autoMap Volume.PricePerGBPerMonth
	pricingToPrice(schema.Pricing) Price
```
Keep in mind this only works if every schema.Pricing -> Price conversion always uses GBPricing. For this specific use case, your solution looks more explicit, so I'm not sure if I'd recommend this.

Reply:
How would the last step work for multiple mappings? For example:
```go
	// goverter:map . Image
	// goverter:map . Volume
	// ...
	PricingFromSchema(schema.Pricing) Pricing

	// goverter:map . PerGBMonth
	imagePricingFromSchema(schema.Pricing) ImagePricing

	// goverter:map . PerGBMonthly
	volumePricingFromSchema(schema.Pricing) VolumePricing

	// goverter:autoMap Image.PricePerGBMonth
	// goverter:autoMap Volume.PricePerGBPerMonth
        // ...
	pricingToPrice(schema.Pricing) Price

```
This would result in a conflict, right?
Also, what about the slices? For some resources (like `PrimaryIP`s) there are different prices per location so they're stored in a slice.

Reply:
Yeah, you're right, this will conflict here. The custom implementation looks best.